scu-de/src/fiskaltrust.Middleware.SCU.DE.SwissbitCloudV2/SwissbitCloudV2SCU.cs

```diff
@@ -0,0 +1,551 @@
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Failed to execute {Operation} - Request: {Request}", nameof(UpdateTransactionAsync), JsonConvert.SerializeObject(request));
+                throw;
+            }
+        }
+
+        public async Task<FinishTransactionResponse> FinishTransactionAsync(FinishTransactionRequest request)
+        {
+            try
+            {
+                if (!await _clientCache.IsClientExistent(request.ClientId))
+                {
+                    throw new Exception($"The client {request.ClientId} is not registered.");
+                }
+
+                var finishTransactionRequest = new TransactionRequestDto()
+                {
+                    ClientId = request.ClientId,
+                    ProcessData = request.ProcessDataBase64,
+                    ProcessType = request.ProcessType,
+                    Number = (int) request.TransactionNumber,
+                };
+                var finishTransactionResponse = await _swissbitCloudV2Provider.TransactionAsync(TransactionType.FinishTransaction, finishTransactionRequest);
+
+                return CreateFinishTransactionResponse(request.ClientId, finishTransactionRequest, finishTransactionResponse);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Failed to execute {Operation} - Request: {Request}", nameof(FinishTransactionAsync), JsonConvert.SerializeObject(request));
+                throw;
+            }
+        }
+
+        public async Task<TseInfo> GetTseInfoAsync()
+        {
+            try
+            {
+                var clientDto = await _swissbitCloudV2Provider.GetClientsAsync().ConfigureAwait(false);
+                var tseResult = await _swissbitCloudV2Provider.GetTseStatusAsync();
+                var startedTransactions = await _swissbitCloudV2Provider.GetStartedTransactionsAsync();
+
+                var bytes = Encoding.ASCII.GetBytes(tseResult.CertificateChain);
+                var cert = new X509Certificate2(bytes);
+
+                var certPublicKey = BitConverter.ToString(cert.GetPublicKey());
+                var certPublicKeyBytes = Encoding.ASCII.GetBytes(certPublicKey);
+
+                var algorithm = cert.SignatureAlgorithm.FriendlyName;
+
+                var tseInfo= new TseInfo
+                {
+                    CurrentNumberOfClients = tseResult.NumberOfRegisteredClients,
+                    CurrentNumberOfStartedTransactions = tseResult.NumberOfStartedTransactions,
+                    SerialNumberOctet = tseResult.SerialNumber,
+                    PublicKeyBase64 = Convert.ToBase64String(certPublicKeyBytes),
+                    FirmwareIdentification = tseResult.SoftwareVersion,
+                    CertificationIdentification = tseResult.CreditClientId,//To ask
```

Comment:
This should be the certification ID from the configuration, i.e. the "BSI-..." string.

Reply:
done